### ISSUE TITLE
Multi-account support

### ISSUE THREAD

**Issue by eragmus** [voisine/breadwallet#274]
I didn't consider the utility of this feature before, but I realized today that having multi-account (multiple wallets) support in breadwallet would be fantastic (see Copay implementation to get an idea of what I mean). The reason is to maintain privacy, by separating addresses used for different purposes ('main' wallet can be for savings, 'spending' wallet can be for trivial day-to-day transactions, 'donation' wallet can be used to post on social media accounts to accept donations). This way, doing basic spending will not run the risk of moving funds around too much and revealing connections between addresses.

Thoughts?

**Comment by voisine** [comment]
I agree that having multiple accounts is useful concept for managing finances, but I don't think we should implement it for the purpose of letting users manually manage their privacy. Very few people will ever understand the underlying bitcoin protocol well enough to manually organize their funds in an effort to combat de-anonymizing network analysis techniques. That's asking way to much of users. I'd rather spend that effort working on automated privacy enhancing techniques like merge avoidance and ultimately things like confidential transactions on sidechains.

**Comment by dabura667** [comment]
I agree with Aaron.

However, the UI could be abstracted away fairly easily:

1. Default one pocket.
2. After receiving at least one transaction, enable a toggle that says "Use multiple pockets" each subsequent pocket can only be used once the previous pocket is used at least once, to help with syncing.
3. When the toggle is active, a small icon appears in the lower left corner of the send/receive panes.
4. Also, while the toggle is active, in small blue letters below the breadwallet / balance it will constantly show "main pocket" ... "pocket 2" ... "pocket 3" etc. depending on the pocket you have chosen.
5. tapping the small icon in the lower left will pop up a swipe menu (similar to the task switcher for iOS) that will swap between pocket "panes"
6. switching pockets can not be done while in "locked" mode (hiding balance) nor can it be done from the Today widget, but if active, the Today widget will show which pocket is active.
7. disabling the "use multiple pockets" toggle will automatically switch back to the main pocket and remove the labels etc.

Adding something like this might require a few considerations like adding the first 20 addresses of pockets 2-10 or something to the bloom filter on syncing.

You know what... the more I write out the feature, the lower the cost-benefit analysis in my brain goes. That's a lot of work for a very negligible feature...

**Comment by voisine** [comment]
I would probably take design cues from apps like Facebook and Twitter for how to make switching and/or creating accounts discoverable and intuitive. Accounts would be virtual within the wallet for the purpose of organizing your own funds, so you could move money between them freely without incurring fees. Nothing to do with bitcoin privacy at all which is managed internally.

**Comment by willnwhite** [comment]
Having separate testnet and mainnet wallets would be handy.

**Comment by mortengh** [comment]
I have an address for me and one for each of my infant children. Currently, without being able to switch between accounts, how do I actually do this in BreadWallet? Will I need to 'start/recover another wallet' all the time? This is way to complicated or will I need 3 iPhones?

**Comment by voisine** [comment]
The app isn't designed for letting you manage someone else's wallet. I would suggest you generate a recovery phrase, copy the receive address, then wipe it, once for each child. Keep the recovery phrase in a safe place to give them when you think they're old enough.

**Comment by mortengh** [comment]
Ok, but can everybody live with only one wallet? It just seems to be a very simple feature to support. It is not that hard to understand and use - if multiple accounts are used a dropbox in 'Settings' and the PIN and you are switched over to the other account. And then BreadWallet would be designed to hold a wallet for each of my 'personalities'.
Is this something that just need to be agreed on or is this a feature that you never think we should have?

**Comment by voisine** [comment]
Right now, the limitation is key management. It's already quite problematic to ask users to write down, not lose, and keep secret one recovery phrase. Doing that for multiple phrases is not something we would ever do. It's already the worst thing about the app.

We could derive multiple wallets from a single recovery phrase, but that has drawbacks like requiring a tx fee to move money between your own wallets, so that's why I think we're more likely to go the route of virtual accounts within the wallet.

That said, pull requests are always welcome! :)

**Comment by Logicwax** [comment]
I actually use "Coins" from the appstore for this purpose.   It seems to be exact re-branded clone of Breadwallet.   I simply use it as a secondary wallet that I use for other finances.
